### ISSUE TITLE
Overriding defaults for postgresql_default_auth_method does not affect postgresql_pg_hba_default

### ISSUE THREAD

**Issue by brainstorm** [ANXS/postgresql#103]
When overriding the following default variables in a local role (mainly setting `md5` instead of `ident`):

```
postgresql_default_auth_method: 'md5'
postgresql_pg_hba_default:
  - { type: local, database: all, user: '{{ postgresql_admin_user }}', address: '', method: '{{ postgresql_default_auth_method }}', comment: '' }
  - { type: local, database: all, user: all, address: '',             method: '{{ postgresql_default_auth_method }}', comment: '"local" is for Unix domain socket connections only' }
  - { type: host,  database: all, user: all, address: '127.0.0.1/32', method: '{{ postgresql_default_auth_method }}', comment: 'IPv4 local connections:' }
  - { type: host,  database: all, user: all, address: '::1/128',      method: '{{ postgresql_default_auth_method }}', comment: 'IPv6 local connections:' }
```

The file present in `/var/lib/pgsql/data/pg_hba.conf` is unaffected. Therefore the following error rises:

```
failed: [10.10.10.9] => {"failed": true}
msg: unable to connect to database: FATAL:  Ident authentication failed for user "postgres"


FATAL: all hosts have already failed -- aborting
```

**Comment by farridav** [comment]
Looking to test this now, If we can reproduce, we will hope to get a fix up shorty, thanks for raising :)

**Comment by farridav** [comment]
I was unable to reproduce locally, but have added a travis test in to see if it fails:

[code here](https://github.com/ANXS/postgresql/compare/farridav/role_based_testing...farridav/md5_auth_test)
[test run here](https://travis-ci.org/ANXS/postgresql/builds/88175587)

In the meantime can you tell me what version of postgres your running against? or even a gist of your play? thanks

**Comment by brainstorm** [comment]
I am running the following playbook:

https://github.com/brainstorm/arvados-ansible

Long story short, you can deploy it via:

`vagrant up workbench && vagrant provision workbench`

I see your test and most probably I'm doing something wrong while running the aforementioned code in:

https://github.com/brainstorm/arvados-ansible/blob/arvados-git/roles/arvados-api/defaults/main.yml

Unless I manually change it via shell/sed:

https://github.com/brainstorm/arvados-ansible/blob/arvados-git/roles/arvados-api/tasks/postgresql.yml#L14

Perhaps you can spot what's wrong there right away? :/

**Comment by farridav** [comment]
Ill try to run again using a centos 6 base box.. perhaps its os specific... if it turns out to be, we might be able to work something in with `ansible_os_family` ..

**Comment by brainstorm** [comment]
Thanks @farridav, that should help us debug what's going on with https://github.com/guillermo-carrasco/arvados-ansible/issues/5 and generalize the postgresql installation.

@guillermo-carrasco

**Comment by farridav** [comment]
@brainstorm I'm looking through your code at https://github.com/brainstorm/arvados-ansible and I cant see ANXS.postgresql used anywhere ? also in your fork I don't see a `VagrantFile`... the only one I can find is in the upstream repo.. and as highlighted earlier, thats using centos ... which currently wont work with this role as we are using `apt` resources in https://github.com/ANXS/postgresql/blob/master/tasks/install.yml#L5 which is tied to ubuntu (Not sure how you got as far as the postgres auth method)... If I've missed/misunderstood something and your able to provide me with a checkout that uses this role and reproduces the error, im happy to take a look...

Otherwise we can raise a feature request to support `centos` ?
 
Cheers

**Comment by brainstorm** [comment]
Oh, sorry @farridav, I had the relevant changes in `arvados-git` and `api` branches. But yeah, I started with ubuntu then ANXS made sense... now we had to switch to CentOS, so that feature request could be very interesting to have indeed!

For now though, I will just stick with the ansible postgresql primitives:

https://github.com/brainstorm/arvados-ansible/blob/arvados-git/roles/arvados-api/tasks/postgresql.yml

But having your role supported by CentOS would be indeed great!
